store/migration/10001__init_schema.sql

```diff
@@ -277,6 +277,8 @@ CREATE TABLE project_member (
     updater_id INTEGER NOT NULL REFERENCES principal (id),
     updated_ts BIGINT NOT NULL DEFAULT (strftime('%s', 'now')),
     project_id INTEGER NOT NULL REFERENCES project (id),
+    -- allowed role_provider are 'BYTEBASE', 'GITLAB_SELF_HOST'.
+    role_provider TEXT NOT NULL,
```

Comment:
where do we store the extra info like when the member is synced and what's the role in the corresponding gitlab project?

Reply:
For now, the only circumstance where a project member would be patched is when we try to modify her role. If the role provider is GitLab, the updatedTs should be the sync timing. Thus, there is no need for an extra field.

As for the second thing you mentioned, Do we need to provide such information to the users? Because users are only allowed to sync the members when they config a corresponding GitLab project to Bytebase, it should be quite straightforward that the project should be the one users had configured.

Reply:
First one is a fair point for now. On the other hand, if we have new patch scenario, then it will become invalid.

For the 2nd point, it's a better UX if we can list what's the corresponding role from the GitLab repo.

From the UI, we can display something like "mapped from Maintainer role, last synced at xxx".  This is useful when the user is confused why a particular member appears in such a role in Bytebase (e.g. user might change the role in GitLab, but Bytebase hasn't re-synced that role for some reason).